frontend/src/components/organizations/OrganizationInfo.vue

```diff
@@ -221,50 +224,146 @@ export default {
+      uploadedDocuments: [],
+      uploadedDocumentsInitialValue: [],
+      documentsToUpload: [],
+      documentsToDelete: [],
+      showUploadDocumentsAlert: false,
     }
   },
-  computed: {
-    ...mapState(useAuthStore, ['userInfo']),
+  watch: {
+    documentsToUpload: {
+      handler(val) {
+        if (val.length === 0 && this.uploadedDocuments.length === 0 && !this.isLoading) {
+          this.showUploadDocumentsAlert = true
+        } else {
+          this.showUploadDocumentsAlert = false
+        }
+      },
+    },
+    'organizationEdit.hasInclusionPolicy': {
+      handler(val, oldVal) {
+        if (oldVal === true && val === false) {
+          this.documentsToDelete = this.uploadedDocuments.map((document) => document.documentId)
+        }
+      },
+    },
   },
   async updated() {
-    this.organizationReadOnly = JSON.parse(JSON.stringify(this.organization))
+    this.loadData()
   },
   methods: {
-    async update() {
-      this.$refs.organizationForm?.validate()
-      if (!this.isFormComplete) {
+    async loadData() {
+      try {
+        this.localLoading = true
+        this.organizationEdit = { ...this.organization, hasInclusionPolicy: this.organization.hasInclusionPolicy || false }
+        if (this.organizationEdit.hasInclusionPolicy) {
+          this.initializeDocuments(await DocumentService.getDocuments(this.organization.organizationId))
+        }
+      } finally {
+        this.localLoading = false
+      }
+    },
+
+    async save() {
+      if (this.organizationEdit?.hasInclusionPolicy && !this.documentsExistOrBeingAdded()) {
+        this.showUploadDocumentsAlert = true
         return
       }
+      this.setLoadingState(true)
       try {
-        this.localLoading = true
-        await OrganizationService.updateOrganization(this.userInfo?.organizationId, this.organizationEdit)
-        this.organizationReadOnly = { ...this.organizationEdit }
+        await OrganizationService.updateOrganization(this.organization.organizationId, this.organizationEdit)
+        await this.processDocuments()
+        this.refreshDocuments()
         this.editMode = false
-        this.setSuccessAlert('Organization information updated successfully')
+        this.showUploadDocumentsAlert = false
+        this.setSuccessAlert('Inclusion Policy updated successfully')
       } catch (error) {
-        this.setFailureAlert('Failed to get your organization information', error)
+        this.setFailureAlert('Failed update Inclusion Policy on Organization: ' + this.organization.organizationId, error)
       } finally {
-        this.localLoading = false
+        this.setLoadingState(false)
       }
     },
 
+    setLoadingState(isLoading) {
+      this.localLoading = isLoading;
+      this.$emit('update:loading', isLoading);
+    },
+
+    async refreshDocuments() {
+      const documents = await DocumentService.getDocuments(this.organization.organizationId);
+      this.initializeDocuments(documents);
```

Comment:
This same code is used on line 257 but without a method call. We should be consistent. I don't think the method is necessary given that it an be done in one line and it calls yet another custom method.

Reply:
You're right, not necessary... i just thought it made the readability better in the save code. Updated.